### DISCUSSION TITLE
web-vitals setup without Vercel Analytics

### DISCUSSION THREAD

**Discussion by jeroenroumen-acc** [vercel/next.js#43516] [Help]
Maybe an uncommon question, but trying to implement Google's `web-vitals` package right now in a Next 12 app and was wondering if anyone has done the same and how to go about it? Putting the `onCLS` etc. functions inside of a `_app.tsx` and running a function in a useEffect technically works, but there's probably a more elegant solution.

Would this be an okay approach for projects that are self hosted or would it be worth it to implement it through Vercel Analytics now that's available.

**Comment by zeckdude** [comment]
We used Sentry for this
